packages/runtime-core/src/apiWatch.ts

```diff
@@ -61,6 +61,9 @@ export type StopHandle = () => void
 
 const invoke = (fn: Function) => fn()
 
+// initial value for watchers to trigger on undefined initial values
+const INITIAL_WATCHER_VALUE = {}
```

Comment:
We could also use a Symbol but it shouldn't change anything since this variable is never exposed to user code

Reply:
hmm, since objects are unique your way is good too. But symbols can help e.g. in debug of Vue itself.

Reply:
> But symbols can help e.g. in debug of Vue itself.

What case are you thinking of?

Symbols are also not supported by all platforms (even though we are using them in quite a few places already). To me, using an object works everywhere while being as simple as using Symbols and I couldn't think of a case where it would fail

Reply:
okay, object wins!